File: spring-web-java/src/main/java/spring/web/java/interfaces/dto/response/ItemResponse.java
package spring.web.java.interfaces.dto.response;

import spring.web.java.domain.model.entity.Item;

public record ItemResponse(
	Long id,
	String name,
	String description,
	int price,
	int quantity
) {
	public ItemResponse(Item item) {
		this(
			item.getId(),
			item.getName(),
			item.getDescription(),
			item.getPrice(),
			item.getQuantity()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/dto/response/ItemResponse.kt
package spring.web.kotlin.dto.response

import spring.web.kotlin.domain.Item

data class ItemResponse(
    val id: Long,
    val name: String,
    val description: String,
    val price: Int,
    val quantity: Int
) {
    constructor(item: Item) : this(
        item.id!!,
        item.name,
        item.description,
        item.price,
        item.quantity
    )
}
